src/server/pfs/pfs.go

```diff
@@ -270,8 +272,17 @@ func (e ErrCommitNotFinished) Error() string {
 	return fmt.Sprintf("commit %v not finished", e.Commit)
 }
 
-func (e ErrCommitNotFinished) GRPCStatus() *status.Status {
-	return status.New(codes.Unavailable, e.Error())
+func (err ErrCommitNotFinished) GRPCStatus() *status.Status {
+	s, sErr := status.New(codes.Unavailable, fmt.Sprintf("commit %v not finished", err.Commit)).WithDetails(&epb.ResourceInfo{
```

Comment:
https://grpc.github.io/grpc/core/md_doc_statuscodes.html

It seems like we should return `FAILED_PRECONDITION` when reading from open commits, and `UNAVAILABLE` when reading from finishing commits.

I don't know if we disambiguate, so this is probably fine.

Reply:
We don’t currently. I’ve opened CORE-1799 to consider disambiguation.